docs/0001-teleport-job-worker.md

```diff
@@ -0,0 +1,185 @@
+---
+authors: Brodie Kurczynski (brodie@kurczynski.org)
+state: draft
+---
+
+# RFD 1 - Teleport Job Worker
+
+## What
+
+A service that runs commands submitted by multiple users. These commands may run concurrently and will provide the users
+with the ability to limit resource (i.e. CPU, memory, and disk IO) usage. The service will consist of three pieces:
+
+* A reusable library implementing the functionality for working with jobs
+* An API server that wraps the functionality of the library
+* A command line interface (CLI) that communicates with the API server
+
+### Required Approvers
+
+* Engineering: (@sclevine && @gavinfrazar) || (@sclevine && @bernardjkim) || (@sclevine && @jimbishopp) || (
+  @gavinfrazar && @bernardjkim) || (@gavinfrazar @jimbishopp) || (@bernardjkim @jimbishopp)
+
+### Security
+
+Requests made to the service will use mTLS for authentication. Specifically:
+
+* TLS using the modern version 1.3
+* The cipher suite will need no configuration, it will be handled by `crypto/tls` (this was updated
+  in [Go 1.17](https://go.dev/doc/go1.17#crypto/tls))
+* X.509 public/private key pairs will be generated using RSA-4096
```

Comment:
For this challenge I think it's ok to just pre-generate the key pairs for client and server, but can you provide explanation for how you will generate all the key pairs necessary to interact with the service, like specific commands and settings you will use?

Preferably as a script (you can link to it instead of copying it inline)

Reply:
Certainly, I'll link the `Makefile` if that works.